front/src/applications/stdcm/utils/addSecondaryCodesToSimilarTrains.ts

```diff
@@ -0,0 +1,31 @@
+import type { PostSimilarTrainsApiResponse } from 'common/api/osrdEditoastApi';
+import type { StdcmPathStep } from 'reducers/osrdconf/types';
+
+import type { SimilarTrainWithSecondaryCode } from '../types';
+
+export const addSecondaryCodesToSimilarTrains = (
+  similarTrains: PostSimilarTrainsApiResponse['similar_trains'],
+  pathSteps: StdcmPathStep[]
+): SimilarTrainWithSecondaryCode[] => {
+  const nameToSecondaryCode = new Map<string, string>();
+
+  pathSteps.forEach((step) => {
+    const loc = step.location;
+    if (loc?.name && loc?.secondary_code) {
+      nameToSecondaryCode.set(loc.name, loc.secondary_code);
+    }
+  });
+
+  return similarTrains.map((similarTrain) => ({
+    train_name: similarTrain.train_name,
+    start_time: new Date(similarTrain.start_time!),
```

Comment:
Hum if a train is not found on this section, then you might have an error here. 
It would be better to test it:
```suggestion
    start_time: similarTrain.start_time ? new Date(similarTrain.start_time) : undefined,
```

Reply:
done